book/listings/gobject_subclassing/2/custom_button/mod.rs

```diff
@@ -0,0 +1,21 @@
+mod imp;
+
+use gtk::glib;
+use gtk::prelude::*;
+
+glib::wrapper! {
+    pub struct CustomButton(ObjectSubclass<imp::CustomButton>)
+        @extends gtk::Button, gtk::Widget;
+}
+
+impl CustomButton {
+    pub fn new() -> Self {
+        glib::Object::new(&[]).expect("Failed to create Button")
+    }
+
+    pub fn with_label(label: &str) -> Self {
+        let button = Self::new();
+        button.set_label(label);
```

Comment:
This is an anti-pattern btw. Better to do

```suggestion
        glib::Object::new(&[("label", label)]).expect("Failed to create Button")
```

Reply:
Thanks, changed it accordingly.